message_passing_simple/greeting_workflow.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+require 'temporalio/workflow'
+require_relative 'call_greeting_service'
+
+module MessagePassingSimple
+  # A workflow that that returns a greeting in one of multiple supported languages.
+  #
+  # It exposes a query to obtain the current language, a signal to approve the workflow so that it is allowed to return
+  # its result, and two updates for changing the current language and receiving the previous language in response.
+  #
+  # One of the update handlers only mutates and returns local workflow state; the other update handler executes an
+  # activity which calls a remote service, giving access to language translations which are not available in local
+  # workflow state.
+  class GreetingWorkflow < Temporalio::Workflow::Definition
+    # This is the equivalent of:
+    #    workflow_query
+    #    def language
+    #        @language
+    #    end
+    workflow_query_attr_reader :language
+
+    def initialize
+      @greetings = { chinese: '你好，世界', english: 'Hello, world' }
+      @language = :english
+    end
+
+    def execute
+      # In addition to waiting for the `approve` signal, we also wait for all handlers to finish. Otherwise, the
+      # workflow might return its result while an async set_language_using_activity update is in progress.
+      Temporalio::Workflow.wait_condition { @approved_for_release && Temporalio::Workflow.all_handlers_finished? }
+      @greetings[@language]
+    end
+
+    workflow_query
+    def languages(input)
+      # A query handler returns a value: it can inspect but must not mutate the Workflow state.
+      if input['include_unsupported']
+        CallGreetingService.greetings.keys.sort
+      else
+        @greetings.keys.sort
+      end
+    end
+
+    workflow_signal
+    def approve(input)
+      # A signal handler mutates the workflow state but cannot return a value.
+      @approved_for_release = true
+      @approver_name = input['name']
+    end
+
+    workflow_update
+    def set_language(new_language) # rubocop:disable Naming/AccessorMethodName
+      # An update handler can mutate the workflow state and return a value.
+      prev = @language.to_sym
+      @language = new_language.to_sym
+      prev
+    end
+
+    workflow_update_validator(:set_language)
+    def validate_set_language(new_language)
+      # In an update validator you raise any exception to reject the update.
+      raise "#{new_language} is not supported" unless @greetings.include?(new_language.to_sym)
+    end
+
+    workflow_update
+    def apply_language_with_lookup(new_language)
+      # This update handler is async, so it can execute an activity.
```

Comment:
I think that comment came from Python and isn't applicable to Ruby

Reply:
Fixed